athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBMetadataHandler.java

```diff
@@ -159,6 +165,19 @@ public DynamoDBMetadataHandler()
         this.tableResolver = new DynamoDBTableResolver(invoker, ddbClient);
     }
 
+    @Override
+    public GetDataSourceCapabilitiesResponse doGetDataSourceCapabilities(BlockAllocator allocator, GetDataSourceCapabilitiesRequest request)
+    {
+        Map<String, List<OptimizationSubType>> capabilities = new HashMap<>();
+        capabilities.putAll(DataSourceOptimizations.SUPPORTS_LIMIT_PUSHDOWN.withSupportedSubTypes(
+            LimitPushdownSubType.ALL
+        ));
+        capabilities.putAll(DataSourceOptimizations.SUPPORTS_FILTER_PUSHDOWN.withSupportedSubTypes(
+            FilterPushdownSubType.ALL
```

Comment:
Is `FilterPushdownSubType.ALL` supposed to mean the same thing for every connector?

If so, how does this work for DDB, which fails for predicates like this (exclusive + inclusive bound):
sql```where somecol > 123 and somecol <= 555```

Reference for the issue:
https://github.com/awslabs/aws-athena-query-federation/pull/375

We can only pass through predicates like the above when both sides are inclusive so that we can translate it to BETWEEN, like:

sql```where somecol >= 123 and somecol <= 555```

gets translated to:

sql```somecol BETWEEN 123 and 555```

Reply:
Good catch, I didn't know about this. I just pushed new changes to further break down filter subtypes and also to remove it from DDB's capabilities due to DDB not being able to actually represent all of the filter types.

Reply:
For posterity, the DDB issue is actually the following:

For tables with both a partition key and a sort key, queries will fail if they specify a partition key comparison and multiple sort key comparisons.

so if I have a ddb table where the partition key is partKey and my sort key is sortKey, the following will fail:

```
where partKey=123 and sortKey > 1234 and sortKey <= 12345 # can't turn into a BETWEEN, has partition key comparison
```

this will succeed:

```
where sortKey > 1234 and sortKey <= 12345 # can't turn into a BETWEEN, but no partititon key, so fine
```

this will succeed:

```
where partKey=123 and sortKey => 1234 and sortKey <= 12345 # can turn into a BETWEEN + has partition key comparison
```